lib/sycamore/sycamore/transforms/base.py

```diff
@@ -0,0 +1,127 @@
+import logging
+from typing import Any, Callable, Iterable, Optional
+
+import numpy as np
+from ray.data import ActorPoolStrategy, Dataset
+
+from sycamore.data import Document, MetadataDocument
+from sycamore.plan_nodes import Node, UnaryNode
+
+
+def _noneOr(a: Any, default: Any) -> Any:
+    if a is None:
+        return default
+    return a
+
+
+def rename(new_function_name: str):
+    def decorator(f):
+        f.__name__ = new_function_name
+        return f
+
+    return decorator
+
+
+class BaseMap(UnaryNode):
+    """
+    BaseMap abstracts away MetadataDocuments from all other transforms.
+    """
+
+    def __init__(
```

Comment:
Once this is used everywhere, does this replace all of the map functions in https://github.com/aryn-ai/sycamore/blob/main/lib/sycamore/sycamore/transforms/map.py?

Reply:
As discussed, probably not. The map functions in map.py become thin wrappers around this class.